File: app/src/main/java/com/xpf/p2p/fragment/MoreFragment.java
package com.xpf.p2p.fragment;

import android.annotation.SuppressLint;
import android.content.Intent;
import android.os.Build;
import android.view.View;
import android.webkit.WebChromeClient;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.loopj.android.http.RequestParams;
import com.xpf.common.base.BaseFragment;
import com.xpf.p2p.R;
import com.xpf.p2p.ui.multilanguage.view.MultiLanguageActivity;


/**
 * Created by xpf on 2016/11/11 :)
 * Function:更多页面(跑马灯效果)
 */

public class MoreFragment extends BaseFragment {

    TextView tvContent;
    TextView tvLanguage;
    WebView webView;
    ProgressBar progressBar;
    private static final String TAG = "MoreFragment";

    @Override
    public int getLayoutId() {
        return R.layout.fragment_more;
    }

    @Override
    protected String getUrl() {
        return null;
    }

    @Override
    protected RequestParams getParams() {
        return null;
    }

    @Override
    protected void initData(String content) {
        tvContent = mView.findViewById(R.id.tv_content);
        tvLanguage = mView.findViewById(R.id.tvLanguage);
        webView = mView.findViewById(R.id.webView);
        progressBar = mView.findViewById(R.id.progressBar);
        tvLanguage.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(mContext, MultiLanguageActivity.class));
            }
        });

        /*// 方式一:
        tvContent.setFocusable(true); // 获取焦点
        tvContent.setFocusableInTouchMode(true);
        tvContent.requestFocus();
        // 方式二:提供TextView的子视图并重写isFocus()方法*/
        initData();
    }

    @SuppressLint("SetJavaScriptEnabled")
    private void initData() {
        webView.getSettings().setJavaScriptEnabled(true);
        // 设置缓存模式
        webView.getSettings().setCacheMode(WebSettings.LOAD_DEFAULT);
        // 开启 DOM storage API 功能
        webView.getSettings().setDomStorageEnabled(true);
        // 解除数据阻止
        webView.getSettings().setBlockNetworkImage(false);
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            webView.getSettings().setMixedContentMode(WebSettings.MIXED_CONTENT_COMPATIBILITY_MODE);
        }
        webView.setWebChromeClient(new WebChromeClient() {
            @Override
            public void onProgressChanged(WebView view, int newProgress) {
                super.onProgressChanged(view, newProgress);
                progressBar.setProgress(newProgress);
                if (newProgress == 100) {
                    progressBar.setVisibility(View.GONE);
                }
            }
        });
        webView.setWebViewClient(new WebViewClient() {
            @Override
            public boolean shouldOverrideUrlLoading(WebView view, String url) {
                view.loadUrl(url);
                return true;
            }
        });

        webView.loadUrl("https://github.com/xinpengfei520/iMoney");
    }
}

File: app/src/main/java/com/xpf/p2p/fragment/MoreFragment.kt
package com.xpf.p2p.fragment

import android.annotation.SuppressLint
import android.content.Intent
import android.net.Uri
import android.widget.Toast
import com.xpf.p2p.R
import com.xpf.p2p.base.BaseFragment
import com.xpf.p2p.databinding.FragmentMoreBinding
import com.xpf.p2p.ui.multilanguage.view.MultiLanguageActivity
import com.xpf.p2p.utils.AppUtil

/**
 * Created by xpf on 2016/11/11 :)
 * Function:更多页面
 */
class MoreFragment : BaseFragment() {

    private var _binding: FragmentMoreBinding? = null
    private val binding get() = _binding!!

    override fun getLayoutId(): Int = R.layout.fragment_more

    override fun getUrl(): String = ""

    override fun getParams(): Map<String, String>? = null

    @SuppressLint("SetTextI18n")
    override fun initData(content: String?) {
        _binding = FragmentMoreBinding.bind(mView!!)

        // 版本号
        binding.tvVersion.text = "V${AppUtil.getAppVersionName(activity!!)}"

        // 多语言
        binding.itemLanguage.setOnClickListener {
            startActivity(Intent(mContext, MultiLanguageActivity::class.java))
        }

        // 消息通知
        binding.itemNotification.setOnClickListener {
            Toast.makeText(activity, "敬请期待", Toast.LENGTH_SHORT).show()
        }

        // 安全中心
        binding.itemSecurity.setOnClickListener {
            Toast.makeText(activity, "敬请期待", Toast.LENGTH_SHORT).show()
        }

        // 清除缓存
        binding.itemClearCache.setOnClickListener {
            Toast.makeText(activity, "缓存已清除", Toast.LENGTH_SHORT).show()
        }

        // 检查更新
        binding.itemCheckUpdate.setOnClickListener {
            Toast.makeText(activity, "当前已是最新版本", Toast.LENGTH_SHORT).show()
        }

        // 关于我们
        binding.itemAbout.setOnClickListener {
            Toast.makeText(activity, "敬请期待", Toast.LENGTH_SHORT).show()
        }

        // GitHub
        binding.itemGithub.setOnClickListener {
            val intent = Intent(Intent.ACTION_VIEW, Uri.parse("https://github.com/xinpengfei520/iMoney"))
            startActivity(intent)
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}
